File: src/main/java/me/muklis/demospringr2dbc/model/Product.java
package me.muklis.demospringr2dbc.model;

import lombok.Data;
import org.springframework.data.annotation.Id;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import java.math.BigDecimal;

@Entity
@Data
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;
    private String name;
    private BigDecimal price;
    private Integer quantity;

}

File: src/main/java/me/muklis/demospringr2dbc/model/Product.kt
package me.muklis.demospringr2dbc.model

import org.springframework.data.annotation.Id
import javax.persistence.Entity

@Entity
class Product {

    @Id
    val id: Int = 0
    val name: String = ""
    val price: Long = 0L
    val quantity: Int = 0

}